src/RemediationRoleSaver.cpp

```diff
@@ -53,22 +83,76 @@ void RemediationSaverBase::selectFilenameAndSaveRole()
 
     try {
         saveToFile(filename);
-        // TODO: if OK - inform the user
+        saveFileOK(filename);
     }
     catch (std::exception& exc)
     {
-        // TODO: if not OK - show error message
-        std::cerr << QObject::tr("Error saving remediation role: %1\n").arg(exc.what()).toUtf8().constData();
+        saveFileError(filename, QString::fromUtf8(exc.what()));
     }
 }
 
-void RemediationSaverBase::saveToFile(const QString& filename)
+
+void RemediationSaverBase::removeFileWhenEmpty(const QString& filename)
+{
+    QFile outputFile(filename);
+    outputFile.open(QIODevice::ReadOnly);
+    int fileSize = outputFile.size();
+    outputFile.close();
+
+    if (fileSize == 0)
+        QFile::remove(filename);
+}
+
+
+void RemediationSaverBase::saveFileOK(const QString& filename)
+{
+    // TODO: if OK - inform the user
+}
+
+
+void RemediationSaverBase::saveFileError(const QString& filename, const QString& errorMsg)
+{
+    removeFileWhenEmpty(filename);
+    const QString completeErrorMessage = QObject::tr("Error saving remediation role to '%2': %1\n").arg(errorMsg, filename);
+    if (mDiagnostics != NULL)
+    {
+        mDiagnostics->errorMessage(completeErrorMessage);
+    }
+    else
+    {
+        std::cerr << completeErrorMessage.toUtf8().constData();
+
+        // TODO: This doesn't work for some reason, even if mParentWindow points to ResultViewer instance
+        QErrorMessage qErrorMsg(mParentWindow);
```

Comment:
because it goes out of scope and is deallocated... but this is a no-go regardless, let's not invent more ways to show errors in SCAP Workbench, let's instead use what we already have